derive/src/relation.rs

```diff
@@ -122,23 +112,50 @@ pub fn expanded_relation_fn(item: &syn::ItemImpl) -> Result<TokenStream, crate::
         })
         .collect::<Result<Vec<ExpandedParams>, crate::error::Error>>()?;
 
-    let (loaders, functions): (Vec<_>, Vec<_>) = expanded_params
+    let relations_parameters: Vec<(String, Option<String>, Option<String>, bool)> = expanded_params
         .iter()
-        .map(
-            |params| -> Result<(TokenStream, TokenStream), crate::error::Error> {
-                let belongs_to = if params.relation_type.to_string().eq("belongs_to") {
-                    Some(params.related_type.to_token_stream().to_string())
-                } else {
-                    None
-                };
+        .map(|params| -> Result<(String, Option<String>, Option<String>, bool), crate::error::Error> {
+            let belongs_to = if params.relation_type.to_string().eq("belongs_to") {
+                Some(params.related_type.to_token_stream().to_string())
+            } else {
+                None
+            };
+
+            let has_many = if params.relation_type.to_string().ne("belongs_to") {
+                Some(params.related_type.to_token_stream().to_string())
+            } else {
+                None
+            };
+
+            let relation_name = params.variant.to_string();
+
+            Ok((relation_name, belongs_to, has_many, false))
+        })
+        .collect::<Result<Vec<_>, crate::error::Error>>()?;
 
-                let has_many = if params.relation_type.to_string().ne("belongs_to") {
-                    Some(params.related_type.to_token_stream().to_string())
-                } else {
-                    None
-                };
+    produce_relations(relations_parameters)
+}
 
-                relation_fn(params.variant.to_string(), belongs_to, has_many)
+pub fn produce_relations(
+    relations_parameters: Vec<(String, Option<String>, Option<String>, bool)>,
+) -> Result<TokenStream, crate::error::Error> {
+    let relations_copy = relations_parameters.clone();
+
+    let reverse_self_references_parameters = relations_copy
+        .into_iter()
+        .filter(|(_, belongs_to, has_one, _)| {
+            belongs_to.eq(&Some("Entity".into())) || has_one.eq(&Some("Entity".into()))
```

Comment:
What if one define the relation as follows?

```rs
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation, seaography::macros::RelationsCompact)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::staff::Entity",
        from = "Column::ReportsToId",
        to = "Column::StaffId",
        on_update = "Restrict",
        on_delete = "Restrict"
    )]
    SelfRef,
}
```

Reply:
belongs_to should equal to "Entity" so we skip this Relation